Watchman.Discord/Areas/Responses/Controllers/ResponsesController.cs

```diff
@@ -26,103 +28,94 @@ public ResponsesController(MessagesServiceFactory messagesServiceFactory, Servic
+            if (command.OnEvent == null || command.Message == null)
             {
                 await messageService.SendResponse(x => x.NotEnoughArguments());
                 return;
             }
-            var response = await this._responsesService.GetResponseByOnEvent(onEvent);
+            var response = await this._responsesService.GetResponseByOnEvent(command.OnEvent);
             if (response == null)
             {
-                await messageService.SendResponse(x => x.ResponseNotFound(contexts, onEvent));
+                await messageService.SendResponse(x => x.ResponseNotFound(contexts, command.OnEvent));
                 return;
             }
-            var responseForThisServer = await this._responsesService.GetResponseByOnEvent(onEvent, contexts.Server.Id);
+            var responseForThisServer = await this._responsesService.GetResponseByOnEvent(command.OnEvent, contexts.Server.Id);
             if (responseForThisServer != null)
             {
-                await messageService.SendResponse(x => x.ResponseAlreadyExists(contexts, onEvent));
+                await messageService.SendResponse(x => x.ResponseAlreadyExists(contexts, command.OnEvent));
                 return;
             }
 
-            var defaultResponse = await this._responsesService.GetResponseByOnEvent(onEvent, DomainResponse.DEFAULT_SERVER_ID);
-            if (defaultResponse.Message == message)
+            var defaultResponse = await this._responsesService.GetResponseByOnEvent(command.OnEvent, DomainResponse.DEFAULT_SERVER_ID);
+            if (defaultResponse.Message == command.Message)
             {
-                await messageService.SendResponse(x => x.ResponseAlreadyExists(contexts, onEvent));
+                await messageService.SendResponse(x => x.ResponseAlreadyExists(contexts, command.OnEvent));
                 return;
             }
 
-            await this._responsesService.AddResponse(onEvent, message, contexts.Server.Id);
-            await messageService.SendResponse(x => x.ResponseHasBeenAdded(contexts, onEvent));
+            await this._responsesService.AddResponse(command.OnEvent, command.Message, contexts.Server.Id);
+            await messageService.SendResponse(x => x.ResponseHasBeenAdded(contexts, command.OnEvent));
         }
 
         [AdminCommand]
-        [DiscordCommand("update response")]
-        public async Task UpdateResponse(DiscordRequest request, Contexts contexts)
+        public async Task UpdateResponse(UpdateResponseCommand command, Contexts contexts)
         {
             var messageService = this._messagesServiceFactory.Create(contexts);
-            var onEvent = request.Arguments.FirstOrDefault(x => x.Name?.ToLowerInvariant() == "onevent")?.Value;
-            var message = request.Arguments.FirstOrDefault(x => x.Name?.ToLowerInvariant() == "message")?.Value;
-            if (onEvent == null || message == null)
+            if (command.OnEvent == null || command.Message == null)
```

Comment:
przecież to nigdy nie będzie nullem

Reply:
a skąd wiesz? wyobraź sobie taką sytuację, że ktoś pisze "-add response -onevent PrintHelp -message" i nagle przez przypadek klika enter zamiast napisać treść dla message. Dla onevent raczej jest to mniej prawdopodobne, ale jednak też możliwe, że to by nie zostało podane. Skoro to ma być profesjonalny bot to powinien mieć zdefiniowane zachowanie dla każdej możliwej sytuacji.

Reply:
ale to chyba w tym przypadku nie zostanie obsłużone jako poprawna komenda, a na etapie parsera już zostanie wyłapana

Reply:
jeśli nie podamy wartości argumentów to też zostanie wyłapana komenda

Reply:
no jak widzisz że jest błąd to go napraw, zamiast tworzyć jakieś dziwne obejścia problemu

lepiej poprawić jednego ifa i mieć spokój wszędzie, niż zawalić aplikacje masą ifów

Reply:
tylko, że nie rozumiem jednej rzeczy, czemu gdy komenda ma niepoprawny format to bot ma nie reagować? Moim zdaniem powinien dać jakiś znak, że coś poszło nie tak. Bo to wygląda tak jakby bot nie wiedział, że taka komenda istnieje.

Reply:
na to jest osobny task

Reply:
które to jest zadanie?

Reply:
https://trello.com/c/qTjt37Ai